src/main/java/duke/io/Parser.java

```diff
@@ -0,0 +1,198 @@
+                day = arr[0];
+            }
+            month = arr[1];
+            try {
+                localDate = LocalDate.of(Integer.parseInt(year),
+                        Integer.parseInt(month), Integer.parseInt(day));
+            } catch (Exception e) {
+                localDate = null;
+            }
+            return localDate;
+        }
+        return null;
+    }
+
+    private String checkTime(String str) {
+        int len = str.length();
+        String lastTwoChars = str.substring(len - 2);
+        String formattedTime = "";
+
+        if (len == 4 && stringToInteger(str) != null) {
+            int time = stringToInteger(str);
+            String period = "am";
+            if (time >= 1200) {
+                period = "pm";
+                formattedTime += (time - 1200);
+            } else {
+                formattedTime += time;
+            }
+            formattedTime = formattedTime + period;
+            return formattedTime;
+        } else if (lastTwoChars.equals("am") || lastTwoChars.equals("pm")) {
+            String [] arr = str.split("\\.");
+            if (arr.length == 1) {
+                return arr[0];
+            } else if (arr.length == 2) {
+                int hour = stringToInteger(arr[0]);
+                int minute = stringToInteger(arr[1].substring(0, arr[1].length() - 2));
+
+                if (0 <= hour && hour <= 12 && 0 <= minute && minute < 60) {
+                    return str;
+                } else {
+                    //Not a valid time so will not be stored
+                    return "";
+                }
+            }
+            return null;
+        }
+        return null;
+    }
+
+    private Integer stringToInteger(String str) {
+        Integer i;
+        try {
+            i = Integer.parseInt(str);
+        } catch (NumberFormatException e) {
+            i = null;
+        }
+        return i;
+    }
+
+    private String formatTime(String t) {
+        String [] splitArr = t.split("\\.");
+
+        if (splitArr.length == 1) {
+            
+            if (t.length() == 3) {
+                //Case: 7pm
+                return t.charAt(0) + ".00" + t.substring(1);
+            } else if(t.length() == 4 && Integer.valueOf(t.substring(0, 2)) < 13) {
+                //Case: 12am
+                return t.substring(0, 2) + ".00" + t.substring(2);
+            } else if (t.length() == 5) {
+                //Case: 732pm 
+                return t.charAt(0) + "." + t.substring(1);
+            } else if (t.length() == 6) {
+                //Case: 1159pm 
+                return t.substring(0, 2) + "." + t.substring(2);
+            } else {
+                return null;
+            }
+        } else {
+            if (splitArr[1].length() == 3) {
+                return splitArr[0] + "." + splitArr[1].substring(0, 1) + "0" + splitArr[1].substring(1,3);
+            } else {
+                return t;
+            }
+        }
+    }
```

Comment:
I think you can consider reorganising these if else statements. The else is quite unnecessary in this case.